src/py/flwr/client/app.py

```diff
@@ -396,13 +399,12 @@ def _on_backoff(retry_state: RetryState) -> None:
                     )
                 else:
                     # Call create_node fn to register node
-                    node_id: Optional[int] = (  # pylint: disable=assignment-from-none
-                        create_node()
-                    )  # pylint: disable=not-callable
-                    if node_id is None:
+                    # and store node_id in state
+                    state.set_node_id(create_node())
```

Comment:
`create_node()` returns `Optional[int]`, so we need to check the return value and exit if node registration fails.